tests/rptest/tests/e2e_iam_role_test.py

```diff
@@ -145,6 +145,69 @@ def test_write(self):
             assert request['response_code'] == 200
 
 
+class ASKRoleFetchTests(EndToEndShadowIndexingBase):
+    def __init__(self, test_context, extra_rp_conf=None):
+        self.iam_server = MockIamRolesServer(test_context,
+                                             'aws_iam_role_mock.py',
+                                             mock_target='aks')
```

Comment:
do we need a test for the azure vm handler as well?

Reply:
yep